test/cluster/test_raft_cluster_features.py

```diff
@@ -84,3 +85,50 @@ async def test_cannot_disable_cluster_feature_after_all_declare_support(manager:
     cql = cql = manager.get_cql()
     hosts = await wait_for_cql_and_get_hosts(cql, servers, time.time() + 60)
     await asyncio.gather(*(wait_for_feature('TEST_ONLY_FEATURE', cql, h, time.time() + 60) for h in hosts))
+
+@pytest.mark.asyncio
+@skip_mode("release", "error injections are not supported in release mode")
+async def test_simulate_upgrade_legacy_to_raft_listener_registration(manager: ManagerClient):
+    """
+    We simulate an upgrade from legacy mode to Raft. Our goal is
+    to make sure that the cluster successfully reaches the state
+    where it can start the upgrade procedure.
+
+    This test effectively reproduces the problem described
+    in scylladb/scylladb#18049.
+    """
+
+    # We need this so that the first logs we wait for appear.
+    cmdline = ["--logger-log-level", "raft_topology=debug"]
```

Comment:
This is unneeded. See `SCYLLA_CMDLINE_OPTIONS` in `scylla_cluster.py`:
```python
SCYLLA_CMDLINE_OPTIONS = [
    ...
    '--logger-log-level', 'raft_topology=debug',
    '--logger-log-level', 'query_processor=debug',
    '--logger-log-level', 'group0_raft_sm=trace',
]
```

I'm mentioning this in case you didn't know that we change some log levels in cluster tests for all nodes. No need to change the test because of such a detail.